course_access_groups/serializers.py

```diff
@@ -1,13 +1,75 @@
-from rest_framework import serializers
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, unicode_literals
 
+from rest_framework import serializers
+from openedx.core.lib.api.serializers import CourseKeyField
 from course_access_groups.models import (
-    CourseAccessGroup,
+    Group,
+    CourseGroup,
+    Membership,
+    Rule,
 )
 
 
 class GroupSerializer(serializers.ModelSerializer):
+    id = serializers.ReadOnlyField()
+    organization_name = serializers.ReadOnlyField(source='organization.name')
+
+    class Meta:
+        model = Group
+        fields = [
+            'id', 'name', 'description', 'organization', 'organization_name',
+        ]
+
+
+class MembershipSerializer(serializers.ModelSerializer):
+    id = serializers.ReadOnlyField()
+    user_email = serializers.ReadOnlyField(source='user.email')
+    user_username = serializers.ReadOnlyField(source='user.username')
+    group_name = serializers.ReadOnlyField(source='group.name')
+    group_description = serializers.ReadOnlyField(source='group.description')
+
+    class Meta:
+        model = Membership
+        fields = [
+            'id',
+            'user',
+            'user_email',
+            'user_username',
+            'group',
+            'group_name',
+            'group_description',
+        ]
+
+
+class RuleSerializer(serializers.ModelSerializer):
+    id = serializers.ReadOnlyField()
+    group_name = serializers.ReadOnlyField(source='group.name')
+
+    class Meta:
+        model = Rule
+        fields = [
+            'id',
+            'name',
+            'domain',
+            'group',
+            'group_name',
+        ]
+
+
+class CourseGroupSerializer(serializers.ModelSerializer):
+    id = serializers.ReadOnlyField()
+    course = CourseKeyField(source='course_id')
```

Comment:
Does this expand to nested data or does it cast the course key as a string?

Reply:
Just a course key.

Reply:
Would you please paste an example as a response so we have a visual?

Reply:
```
GET /group-courses/

{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 1,
            "course": "course-v1:org1+course2+202",
            "course_name": "My course",
            "group": 1,
            "group_name": "Fa So La Si University"
        }
    ]
}
```